core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -387,6 +397,46 @@ public String delResource(final ServiceContext context, final String metadataUui
         return String.format("Unable to remove resource '%s'.", resourceId);
     }
 
+    @Override
+    public void copyResources(ServiceContext context, String sourceUuid, String targetUuid, MetadataResourceVisibility metadataResourceVisibility) throws Exception {
+        final int sourceMetadataId = canDownload(context, sourceUuid, metadataResourceVisibility, false);
+        final String sourceResourceTypeDir = getMetadataDir(context, sourceMetadataId) + CMISConfiguration.getFolderDelimiter() + metadataResourceVisibility.toString();
+        try {
+            Folder sourceParentFolder = (Folder) CMISConfiguration.getClient().getObjectByPath(sourceResourceTypeDir);
+
+            Map<String, Document> sourceDocumentMap = getCmisObjectMap(sourceParentFolder, null);
+            for (Map.Entry<String, Document> sourceEntry : sourceDocumentMap.entrySet()) {
+                Document sourceDocument = sourceEntry.getValue();
+                if (sourceDocument instanceof Document) {
+                    Map<String, Object> sourceProperties = getSecondaryProperties(sourceDocument);
+                    putResource(context, targetUuid, sourceDocument.getName(), sourceDocument.getContentStream().getStream(), null, metadataResourceVisibility, true, sourceProperties);
```

Comment:
Have you tried sourceDocument.copy(targetfolder)? 
I have not tested but I believe this may be quicker when copying large files. I believe it would prevent having to send the content from the content server to the application server and then back to the content server.  I would expect the copying to take place on the content server.  Also it may copy the properties as well?  I have not tested.

Reply:
Also not that there are regular properties as well that need to be copied.  Not all properties are secondary type properties.  So we will have to ensure that those are copied as well.

Reply:
I tried everything you pointed out here. If I do document.copy, it will only create new document. If the document is there and we are adding a new version. CMIS will fail.

Also I attempt to put all properties together, I will always get unknown CMIS error. I looked at all other properties, some of them are related to versioning, and it should not passed at all. Should let CMIS and backend content server to handle versioning thing. Really there is very limited properties we can touch and it's pretty dangerous to touch those unknown properties except common one line NAME, LAST_MODIFIED.